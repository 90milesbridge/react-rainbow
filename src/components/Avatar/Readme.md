avatar base

    <div className="slds-p-vertical_large slds-p-left_medium slds-grid slds-media_center slds-wrap">
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar1.jpg"
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                size="large" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar2.jpg"
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                size="medium" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar3.jpg"
                assistiveText="Jonh Doe"
                title="Jonh Doe"
                size="small" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar2.jpg"
                assistiveText="Jane Doe"
                title="Jane Doe"
                size="x-small" />

        </div>
    </div>


avatar circle

    <div className="slds-p-vertical_large slds-p-left_medium slds-grid slds-media_center slds-wrap">
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar1.jpg"
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                size="large"
                variant="circle" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar2.jpg"
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                size="medium"
                variant="circle" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar3.jpg"
                title="Jonh Doe"
                size="small"
                variant="circle" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/avatar2.jpg"
                assistiveText="Jane Doe"
                title="Jane Doe"
                size="x-small"
                variant="circle" />

        </div>
    </div>


fallback user icon

    <div className="slds-p-vertical_x-large slds-p-left_medium slds-grid slds-media_center slds-wrap">
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jose Leandro"
                title="Jose Leandro" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                variant="circle" />

        </div>

    </div>


fallback icon

    <div className="slds-p-vertical_x-large slds-p-left_medium slds-grid slds-media_center slds-wrap">
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                iconName="action:call" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                iconName="custom:custom1" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                iconName="doctype:ai" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                iconName="standard:account" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jose Leandro"
                title="Jose Leandro"
                iconName="utility:activity" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                variant="circle"
                iconName="action:call" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                variant="circle"
                iconName="custom:custom1" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                variant="circle"
                iconName="doctype:ai" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                variant="circle"
                iconName="standard:account" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Tahimi Leon"
                title="Tahimi Leon"
                variant="circle"
                iconName="utility:activity" />

        </div>

    </div>


fallback user initials

    <div className="slds-p-vertical_x-large slds-p-left_medium slds-grid slds-media_center slds-wrap">
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/wrong.jpg"
                assistiveText="Jane Doe"
                initials="JD"
                title="Jane Doe" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jane Doe"
                initials="JD"
                title="Jane Doe"
                variant="circle" />

        </div>
    </div>


fallback user initials inverse

    const avatarContainerStyles = {
        backgroundColor: '#16325c',
        borderRadius: '0.25rem',
    };

    <div className="slds-p-vertical_x-large slds-p-left_medium slds-grid slds-media_center" style={avatarContainerStyles}>
        <div className="slds-m-horizontal_medium">
            <Avatar
                src="images/wrong.jpg"
                assistiveText="Jane Doe"
                initials="TL"
                title="Tahimi Leon"
                initialsVariant="inverse" />

        </div>
        <div className="slds-m-horizontal_medium">
            <Avatar
                assistiveText="Jane Doe"
                initials="LT"
                title="Leandro Torres"
                variant="circle"
                initialsVariant="inverse" />

        </div>
    </div>
