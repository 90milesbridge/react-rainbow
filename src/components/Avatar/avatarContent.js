import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

const INITIALS_REGEX = /^[A-Z][A-Z]$/;

export default class AvatarContent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            imageFailed: false,
        };

        this.handleImageError = this.handleImageError.bind(this);
    }

    handleImageError() {
        this.setState({ imageFailed: true });
    }

    render() {
        const {
            src,
            initials,
            iconName,
            alt,
            title,
        } = this.props;
        const { imageFailed } = this.state;
        const isInitialsValid = INITIALS_REGEX.test(initials);
        if (src && !imageFailed) {
            return <img alt={alt} src={src} onError={this.handleImageError} title={title} />;
        } else if (isInitialsValid) {
            return (
                <abbr
                    className="slds-avatar__initials slds-icon-standard-user"
                    title={title}>

                    {initials}
                </abbr>
            );
        }
        return <Icon iconName={iconName} size="medium" />;
    }
}

AvatarContent.propTypes = {
    alt: PropTypes.string,
    src: PropTypes.string,
    initials: PropTypes.string,
    iconName: PropTypes.string.isRequired,
    title: PropTypes.string,
};

AvatarContent.defaultProps = {
    alt: undefined,
    src: undefined,
    initials: undefined,
    title: undefined,
};
