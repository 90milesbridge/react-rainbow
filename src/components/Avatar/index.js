/* eslint-disable react/require-default-props */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';

const INITIALS_REGEX = /^[A-Z][A-Z]$/;

export default class Avatar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            imageFailed: false,
        };

        this.handleImageError = this.handleImageError.bind(this);
    }

    getContainerClassNames() {
        const { className, size, variant } = this.props;

        return classnames('slds-avatar', {
            'slds-avatar_circle': variant,
            'slds-avatar_large': size === 'large',
            'slds-avatar_medium': size === 'medium',
            'slds-avatar_small': size === 'small',
            'slds-avatar_x-small': size === 'x-small',
        }, className);
    }

    handleImageError() {
        this.setState({ imageFailed: true });
    }

    renderAvatar() {
        const {
            src,
            initials,
            iconName,
            alt,
            style,
            title,
        } = this.props;
        const { imageFailed } = this.state;
        const isInitialsValid = INITIALS_REGEX.test(initials);

        if (src && !imageFailed) {
            return <img alt={alt} src={src} onError={this.handleImageError} title={title} />;
        } else if (isInitialsValid) {
            return (
                <abbr className="slds-avatar__initials slds-icon-standard-user"
                      style={style}
                      title={title}>
                    { initials }
                </abbr>
            );
        }
        return <Icon iconName={iconName} size="medium" variant="bare" />;
    }

    render() {
        const { style } = this.props;
        return (
            <span className={this.getContainerClassNames()} style={style}>
                {this.renderAvatar()}
            </span>
        );
    }
}

Avatar.propTypes = {
    /** Avatar description. It usually is a person name. Can be used only if
     valid src is available */
    alt: PropTypes.string,
    /** Is the avatar source path. Its value is the relative URL to the image.
     It take precedence over the initials and icon */
    src: PropTypes.string,
    /** If set to true make the avatar a circle. */
    variant: PropTypes.bool,
    /** Class for custom styles */
    className: PropTypes.string,
    /** The user initials. It only can have two letters (the first name and last name
     first letter in upper case). It take precedence over the icon */
    initials: PropTypes.string,
    /** The icon name. It have the less precedence. It take the following format:
     ‘sprite name:icon name’ e.g. ‘utility:add’ */
    iconName: PropTypes.string,
    /** The icon size. */
    size: PropTypes.oneOf(['large', 'medium', 'small', 'x-small']),
    /** Object with the custom styles. The properties must be in camelCase naming
    convention (e.g. { backgroundColor: green }) */
    style: PropTypes.object,
    /** This is a description that is showed when a user hover the avatar */
    title: PropTypes.string,
};

Avatar.defaultProps = {
    iconName: 'standard:user',
    size: 'medium',
    variant: false,
};
