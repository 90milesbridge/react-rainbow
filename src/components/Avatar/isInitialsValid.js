export default function isInitialsValid(initials) {
    const INITIALS_REGEX = /^[A-Z][A-Za-z]$/;
    return INITIALS_REGEX.test(initials);
}
