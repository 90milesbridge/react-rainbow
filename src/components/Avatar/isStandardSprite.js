const STANDARD_REGEX = /^standard:.+/;

export default function isStandardSprite(iconName) {
    return STANDARD_REGEX.test(iconName);
}
