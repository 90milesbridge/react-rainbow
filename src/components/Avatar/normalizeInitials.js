export default function normalizeInitials(initials) {
    if (initials.length === 1) {
        return initials;
    }
    return initials.slice(0, 2);
}
