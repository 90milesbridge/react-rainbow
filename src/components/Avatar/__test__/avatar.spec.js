import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Avatar from './../index';

describe('<Avatar/>', () => {
    it('should pass assistiveText to the prop text of AssistiveText component', () => {
        const component = mount(
            <Avatar initials="JD" assistiveText="for screen readers" />,
        );
        expect(component.find('AssistiveText').prop('text')).toBe('for screen readers');
    });
    it('should have the right class names', () => {
        const component = renderer.create(
            <Avatar />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when variant is circle and size is x-small', () => {
        const component = renderer.create(
            <Avatar variant="circle" size="x-small" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when size is small', () => {
        const component = renderer.create(
            <Avatar size="small" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when size is medium', () => {
        const component = renderer.create(
            <Avatar size="medium" />,
        );
        expect(component).toMatchSnapshot();
    });
    it('should have the right class names when size is large', () => {
        const component = renderer.create(
            <Avatar size="large" />,
        );
        expect(component).toMatchSnapshot();
    });
});
