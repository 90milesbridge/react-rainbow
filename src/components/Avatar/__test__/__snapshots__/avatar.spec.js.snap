// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> should have the right class names 1`] = `
<span
  className="slds-avatar slds-avatar_medium"
  style={Object {}}
>
  <span
    className="slds-icon_container slds-icon-standard-user"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="slds-icon"
      style={Object {}}
    >
      <use
        xlinkHref="/icons/standard-sprite/svg/symbols.svg#user"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
</span>
`;

exports[`<Avatar/> should have the right class names when size is large 1`] = `
<span
  className="slds-avatar slds-avatar_large"
  style={Object {}}
>
  <span
    className="slds-icon_container slds-icon-standard-user"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="slds-icon"
      style={Object {}}
    >
      <use
        xlinkHref="/icons/standard-sprite/svg/symbols.svg#user"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
</span>
`;

exports[`<Avatar/> should have the right class names when size is medium 1`] = `
<span
  className="slds-avatar slds-avatar_medium"
  style={Object {}}
>
  <span
    className="slds-icon_container slds-icon-standard-user"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="slds-icon"
      style={Object {}}
    >
      <use
        xlinkHref="/icons/standard-sprite/svg/symbols.svg#user"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
</span>
`;

exports[`<Avatar/> should have the right class names when size is small 1`] = `
<span
  className="slds-avatar slds-avatar_small"
  style={Object {}}
>
  <span
    className="slds-icon_container slds-icon-standard-user"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="slds-icon"
      style={Object {}}
    >
      <use
        xlinkHref="/icons/standard-sprite/svg/symbols.svg#user"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
</span>
`;

exports[`<Avatar/> should have the right class names when variant is circle and size is x-small 1`] = `
<span
  className="slds-avatar slds-avatar_circle slds-avatar_x-small"
  style={Object {}}
>
  <span
    className="slds-icon_container slds-icon-standard-user"
    style={Object {}}
  >
    <svg
      aria-hidden={true}
      className="slds-icon"
      style={Object {}}
    >
      <use
        xlinkHref="/icons/standard-sprite/svg/symbols.svg#user"
        xmlnsXlink="http://www.w3.org/1999/xlink"
      />
    </svg>
  </span>
</span>
`;
